## 📋 Estrutura do Relatório

1. **Trilha 1: Crescimento e Receita** — evolução temporal, análise geográfica e mix de produtos
2. **Trilha 2: Comportamento e Pagamentos** — meios de pagamento e parcelamento
3. **Trilha 3: Satisfação do Cliente** — notas, logística e tempo de resposta
4. **Trilha 4: Retenção e Recompra** — comportamento de recompra e lifetime value
5. **Trilha 5: Oportunidades e Recomendações** — riscos, recomendações e conclusão

**Pergunta norteadora:** Como a Olist pode sustentar seu crescimento e aumentar a receita mantendo altos níveis de satisfação dos clientes e eficiência operacional?

### Bibliotecas necessárias para fazer o Tech Challenge

In [66]:
# Libs - bibliotecas importantes

# Manipulação dos dados
import pandas as pd
import numpy as np

# Visualização dos dados
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

In [67]:
# Ler as Bases  de dados - Olist

df_items = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_order_items_dataset.csv")
df_customers = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_customers_dataset.csv")
df_payments = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_order_payments_dataset.csv")
df_reviews = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_order_reviews_dataset.csv")
df_products = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_products_dataset.csv")
df_sellers = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_sellers_dataset.csv")
df_orders = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_orders_dataset.csv")
df_geolocations = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/olist_geolocation_dataset.csv")
df_product_category_name_translation = pd.read_csv("/Users/lucas.chaves/Desktop/Pós Data Analytics/Tech Challenge/product_category_name_translation.csv")

In [68]:
# visualização das tabelas

print(' ITEMS ')
display(df_items.head())

print(' CUSTOMERS ')
display(df_customers.head())

print(' PAYMENTS ')
display(df_payments.head())

print(' REVIEWS ')
display(df_reviews.head())

print(' PRODUCTS ')
display(df_products.head())

print(' SELLERS ')
display(df_sellers.head())

print(' ORDERS ')
display(df_orders.head())

print(' GEOLOCATIONS ')
display(df_geolocations.head())

print(' PRODUCT CATEGORY TRANSLATION ')
display(df_product_category_name_translation.head())

 ITEMS 


,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14


 CUSTOMERS 


,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


 PAYMENTS 


,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


 REVIEWS 


,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
0,7bc2406110b926393aa56f80a40eba40,73fc7af87114b39712e6da79b0a377eb,4,NaN,NaN,2018-01-18 00:00:00,2018-01-18 21:46:59
1,80e641a11e56f04c1ad469d5645fdfde,a548910a1c6147796b98fdf73dbeba33,5,NaN,NaN,2018-03-10 00:00:00,2018-03-11 03:05:13
2,228ce5500dc1d8e020d8d1322874b6f0,f9e4b658b201a9f2ecdecbb34bed034b,5,NaN,NaN,2018-02-17 00:00:00,2018-02-18 14:36:24
3,e64fb393e7b32834bb789ff8bb30750e,658677c97b385a9be170737859d3511b,5,NaN,Recebi bem antes do prazo estipulado.,2017-04-21 00:00:00,2017-04-21 22:02:06
4,f7c4243c7fe1938f181bec41a392bdeb,8e6bfb81e283fa7e4f11123a3fb894f1,5,NaN,Parabéns lojas lannister adorei comprar pela I...,2018-03-01 00:00:00,2018-03-02 10:26:53


 PRODUCTS 


,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0


 SELLERS 


,seller_id,seller_zip_code_prefix,seller_city,seller_state
0,3442f8959a84dea7ee197c632cb2df15,13023,campinas,SP
1,d1b65fc7debc3361ea86b5f14c68d2e2,13844,mogi guacu,SP
2,ce3ad9de960102d0677a81f5d0bb7b2d,20031,rio de janeiro,RJ
3,c0f3eea2e14555b6faeea3dd58c1b1c3,4195,sao paulo,SP
4,51a04a8a6bdcb23deccc82b0b80742cf,12914,braganca paulista,SP


 ORDERS 


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00


 GEOLOCATIONS 


,geolocation_zip_code_prefix,geolocation_lat,geolocation_lng,geolocation_city,geolocation_state
0,1037,-23.545621,-46.639292,sao paulo,SP
1,1046,-23.546081,-46.644820,sao paulo,SP
2,1046,-23.546129,-46.642951,sao paulo,SP
3,1041,-23.544392,-46.639499,sao paulo,SP
4,1035,-23.541578,-46.641607,sao paulo,SP


 PRODUCT CATEGORY TRANSLATION 


,product_category_name,product_category_name_english
0,beleza_saude,health_beauty
1,informatica_acessorios,computers_accessories
2,automotivo,auto
3,cama_mesa_banho,bed_bath_table
4,moveis_decoracao,furniture_decor


In [69]:
# Verificando os tipos de dados

print(' TIPOS DE DADOS ')
print(' ITEMS ')
print(df_items.dtypes)
print(' CUSTOMERS ')
print(df_customers.dtypes)
print(' PAYMENTS ')
print(df_payments.dtypes)
print(' REVIEWS ')
print(df_reviews.dtypes)
print(' PRODUCTS ')
print(df_products.dtypes)
print(' SELLERS ')
print(df_sellers.dtypes)
print(' ORDERS ')
print(df_orders.dtypes)
print(' GEOLOCATIONS ')
print(df_geolocations.dtypes)
print(' PRODUCT CATEGORY TRANSLATION ')
print(df_product_category_name_translation.dtypes)

 TIPOS DE DADOS 
 ITEMS 
order_id                   str
order_item_id            int64
product_id                 str
seller_id                  str
shipping_limit_date        str
price                  float64
freight_value          float64
dtype: object
 CUSTOMERS 
customer_id                   str
customer_unique_id            str
customer_zip_code_prefix    int64
customer_city                 str
customer_state                str
dtype: object
 PAYMENTS 
order_id                    str
payment_sequential        int64
payment_type                str
payment_installments      int64
payment_value           float64
dtype: object
 REVIEWS 
review_id                    str
order_id                     str
review_score               int64
review_comment_title         str
review_comment_message       str
review_creation_date         str
review_answer_timestamp      str
dtype: object
 PRODUCTS 
product_id                        str
product_category_name             str
product_name_lenght  

In [70]:
# Verificando valores nulos em cada tabela

tabelas = {
    'Items': df_items,
    'Customers': df_customers,
    'Payments': df_payments,
    'Reviews': df_reviews,
    'Products': df_products,
    'Sellers': df_sellers,
    'Orders': df_orders,
    'Geolocations': df_geolocations,
    'Product Category Translation': df_product_category_name_translation
}

for nome, df in tabelas.items():
    print(f'\n {nome} ')
    display(df.isnull().sum())


 Items 


order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64


 Customers 


customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64


 Payments 


order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64


 Reviews 


review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64


 Products 


product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64


 Sellers 


seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64


 Orders 


order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64


 Geolocations 


geolocation_zip_code_prefix    0
geolocation_lat                0
geolocation_lng                0
geolocation_city               0
geolocation_state              0
dtype: int64


 Product Category Translation 


product_category_name            0
product_category_name_english    0
dtype: int64

# Crescimento e Receita


In [71]:
# Junta items com orders pra ter preço + data
df_vendas = pd.merge(
    df_items,
    df_orders,
    on='order_id',
    how='inner'
)

# Confere o tamanho
print(f'df_items: {df_items.shape[0]:,} linhas')
print(f'df_orders: {df_orders.shape[0]:,} linhas')
print(f'df_vendas: {df_vendas.shape[0]:,} linhas')

# Vê as primeiras linhas
df_vendas.head()

df_items: 112,650 linhas
df_orders: 99,441 linhas
df_vendas: 112,650 linhas


,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29,3ce436f183e68e07877b285a838db11a,delivered,2017-09-13 08:59:02,2017-09-13 09:45:35,2017-09-19 18:34:16,2017-09-20 23:43:48,2017-09-29 00:00:00
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93,f6dd3ec061db4e3987629fe6b26e5cce,delivered,2017-04-26 10:53:06,2017-04-26 11:05:13,2017-05-04 14:35:00,2017-05-12 16:04:24,2017-05-15 00:00:00
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87,6489ae5e4333f3693df5ad4372dab6d3,delivered,2018-01-14 14:33:31,2018-01-14 14:48:30,2018-01-16 12:36:48,2018-01-22 13:19:16,2018-02-05 00:00:00
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79,d4eb9395c8c0431ee92fce09860c5a06,delivered,2018-08-08 10:00:35,2018-08-08 10:10:18,2018-08-10 13:28:00,2018-08-14 13:32:39,2018-08-20 00:00:00
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14,58dbd0b2d70206bf40e62cd34e84d795,delivered,2017-02-04 13:57:51,2017-02-04 14:10:13,2017-02-16 09:46:09,2017-03-01 16:42:31,2017-03-17 00:00:00


In [72]:
# Verifica os status dos pedidos

df_vendas['order_status'].value_counts()

order_status
delivered      110197
shipped          1185
canceled          542
invoiced          359
processing        357
unavailable         7
approved            3
Name: count, dtype: int64

In [73]:
# Valores antes do filtro
total_itens = df_vendas.shape[0]
print(f'Antes de filtrar por tipo Delivered: {total_itens:,} linhas')

# Cria um novo DataFrame só com entregues (preserva o df_vendas original)
df_vendas_entregues = df_vendas[df_vendas['order_status'] == 'delivered']

# Tamanho depois do filtro
itens_entregues = df_vendas_entregues.shape[0]
print(f'Depois de filtrar por tipo Delivered: {itens_entregues:,} linhas')

# Percentual de entrega
percentual_entrega = (itens_entregues / total_itens) * 100
print(f'Percentual de entrega: {percentual_entrega:.2f}%')

Antes de filtrar por tipo Delivered: 112,650 linhas
Depois de filtrar por tipo Delivered: 110,197 linhas
Percentual de entrega: 97.82%


In [74]:
# Conta os itens cancelados
itens_cancelados = df_vendas[df_vendas['order_status'] == 'canceled'].shape[0]

# Calcula o percentual
percentual_cancelamento = (itens_cancelados / total_itens) * 100

print(f'Itens cancelados: {itens_cancelados:,}')
print(f'Percentual de cancelamento: {percentual_cancelamento:.2f}%')

Itens cancelados: 542
Percentual de cancelamento: 0.48%


Insight pro relatório

"Do total de itens vendidos no período, 97,8% foram efetivamente entregues, indicando uma operação logística com alta taxa de conclusão. Apenas 0,48% dos itens foram cancelados."

In [75]:
# Antes da conversão - confere o tipo da coluna
print('Tipo ANTES da conversão:', df_vendas_entregues['order_purchase_timestamp'].dtype)

# Converte a coluna pra tipo datetime
df_vendas_entregues['order_purchase_timestamp'] = pd.to_datetime(df_vendas_entregues['order_purchase_timestamp'])

# Depois da conversão - confere de novo
print('Tipo DEPOIS da conversão:', df_vendas_entregues['order_purchase_timestamp'].dtype)

Tipo ANTES da conversão: str
Tipo DEPOIS da conversão: datetime64[us]


In [76]:
# Cria coluna nova com formato ano-mês
df_vendas_entregues['ano_mes'] = df_vendas_entregues['order_purchase_timestamp'].dt.to_period('M')

# Confere o resultado
df_vendas_entregues[['order_purchase_timestamp', 'ano_mes']].head()

,order_purchase_timestamp,ano_mes
0,2017-09-13 08:59:02,2017-09
1,2017-04-26 10:53:06,2017-04
2,2018-01-14 14:33:31,2018-01
3,2018-08-08 10:00:35,2018-08
4,2017-02-04 13:57:51,2017-02


In [77]:
# Agrupa por mês e calcula as métricas
evolucao = df_vendas_entregues.groupby('ano_mes').agg(
    qtd_pedidos=('order_id', 'nunique'),
    receita_total=('price', 'sum')
).reset_index()

# Calcula ticket médio (receita ÷ pedidos)
evolucao['ticket_medio'] = evolucao['receita_total'] / evolucao['qtd_pedidos']

# Ordena por mês (do mais antigo pro mais novo)
evolucao = evolucao.sort_values('ano_mes').reset_index(drop=True)

# Mostra o resultado
display(evolucao)

,ano_mes,qtd_pedidos,receita_total,ticket_medio
0,2016-09,1,134.97,134.970000
1,2016-10,265,40325.11,152.170226
2,2016-12,1,10.90,10.900000
3,2017-01,750,111798.36,149.064480
4,2017-02,1653,234223.40,141.695947
5,2017-03,2546,359198.85,141.083602
6,2017-04,2303,340669.68,147.924307
7,2017-05,3546,489338.25,137.997250
8,2017-06,3135,421923.37,134.584807
9,2017-07,3872,481604.52,124.381333


In [78]:
# Filtra só a partir de janeiro/2017
evolucao_mensal = evolucao[evolucao['ano_mes'] >= '2017-01'].reset_index(drop=True)

display(evolucao_mensal)

,ano_mes,qtd_pedidos,receita_total,ticket_medio
0,2017-01,750,111798.36,149.064480
1,2017-02,1653,234223.40,141.695947
2,2017-03,2546,359198.85,141.083602
3,2017-04,2303,340669.68,147.924307
4,2017-05,3546,489338.25,137.997250
5,2017-06,3135,421923.37,134.584807
6,2017-07,3872,481604.52,124.381333
7,2017-08,4193,554699.70,132.291844
8,2017-09,4150,607399.67,146.361366
9,2017-10,4478,648247.65,144.762762


**Sobre o período analisado:** Optamos por analisar a partir de janeiro/2017, já que os meses anteriores tiveram volumes muito baixos (sendo que setembro e dezembro de 2016 registraram apenas 1 pedido cada).

In [79]:
import plotly.graph_objects as go

# Converte ano_mes pra texto (se ainda não foi convertido)
evolucao_mensal['ano_mes_str'] = evolucao_mensal['ano_mes'].astype(str)

# Cria a figura vazia
fig_pedidos = go.Figure()

# Adiciona a linha de pedidos (azul-marinho)
fig_pedidos.add_trace(
    go.Scatter(
        x=evolucao_mensal['ano_mes_str'],
        y=evolucao_mensal['qtd_pedidos'],
        name='Pedidos',
        mode='lines+markers',
        line=dict(color='#000A53', width=2),
        marker=dict(size=8, symbol='circle')
    )
)

# Configura o layout no mesmo padrão do gráfico combinado
fig_pedidos.update_layout(
    title=dict(
        text='Evolução Mensal de Pedidos (Jan/2017 - Ago/2018)',
        x=0.5,
        xanchor='center'
    ),
    xaxis=dict(title='Mês', tickangle=-45),
    yaxis=dict(
        title=dict(text='Quantidade de Pedidos', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53')
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_pedidos.show()

**Evolução de pedidos:**

A quantidade de pedidos cresceu de forma consistente em 2017, partindo de 750 em janeiro e chegando a 7.289 em novembro. O pico de novembro coincide com a Black Friday. Em 2018, o volume estabilizou na faixa de 6.000 a 7.000 pedidos por mês.

In [80]:
# Calcula a variação percentual mês a mês
evolucao_mensal['var_pct_pedidos'] = evolucao_mensal['qtd_pedidos'].pct_change() * 100
evolucao_mensal['var_pct_receita'] = evolucao_mensal['receita_total'].pct_change() * 100

# Mostra a tabela atualizada
display(evolucao_mensal)

,ano_mes,qtd_pedidos,receita_total,ticket_medio,ano_mes_str,var_pct_pedidos,var_pct_receita
0,2017-01,750,111798.36,149.064480,2017-01,NaN,NaN
1,2017-02,1653,234223.40,141.695947,2017-02,120.400000,109.505220
2,2017-03,2546,359198.85,141.083602,2017-03,54.022989,53.357372
3,2017-04,2303,340669.68,147.924307,2017-04,-9.544383,-5.158471
4,2017-05,3546,489338.25,137.997250,2017-05,53.973079,43.640094
5,2017-06,3135,421923.37,134.584807,2017-06,-11.590525,-13.776744
6,2017-07,3872,481604.52,124.381333,2017-07,23.508772,14.145021
7,2017-08,4193,554699.70,132.291844,2017-08,8.290289,15.177428
8,2017-09,4150,607399.67,146.361366,2017-09,-1.025519,9.500631
9,2017-10,4478,648247.65,144.762762,2017-10,7.903614,6.725058


**Ritmo de crescimento dos pedidos:**

Os primeiros meses de 2017 tiveram crescimento expressivo, com fevereiro registrando +120%. A Black Friday (nov/2017) trouxe outro salto importante: +63% em um único mês. Em 2018, as variações se estabilizaram entre -10% e +10%, indicando que a operação encontrou um patamar de demanda mais constante.

In [ ]:
import plotly.graph_objects as go

# Cria a figura vazia
fig_receita = go.Figure()

# Adiciona a linha de receita (vinho)
fig_receita.add_trace(
    go.Scatter(
        x=evolucao_mensal['ano_mes_str'],
        y=evolucao_mensal['receita_total'],
        name='Receita (R$)',
        mode='lines+markers',
        line=dict(color='#9b1c31', width=2, dash='dash'),
        marker=dict(size=8, symbol='square')
    )
)

fig_receita.update_layout(
    title=dict(
        text='Evolução Mensal de Receita (Jan/2017 - Ago/2018)',
        x=0.5,
        xanchor='center'
    ),
    xaxis=dict(title='Mês', tickangle=-45),
    yaxis=dict(
        title=dict(text='Receita Total (R$)', font=dict(color='#9b1c31')),
        tickfont=dict(color='#9b1c31'),
        tickprefix='R$ ',
        tickformat=',.0f'
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_receita.show()

In [132]:
import plotly.graph_objects as go

fig_combinado = go.Figure()

# Linha de pedidos (azul-marinho)
fig_combinado.add_trace(
    go.Scatter(
        x=evolucao_mensal['ano_mes_str'],
        y=evolucao_mensal['qtd_pedidos'],
        name='Pedidos',
        mode='lines+markers',
        line=dict(color='#000A53', width=2),
        marker=dict(size=8, symbol='circle'),
        yaxis='y1'
    )
)

# Linha de receita (vinho)
fig_combinado.add_trace(
    go.Scatter(
        x=evolucao_mensal['ano_mes_str'],
        y=evolucao_mensal['receita_total'],
        name='Receita (R$)',
        mode='lines+markers',
        line=dict(color='#9b1c31', width=2, dash='dash'),
        marker=dict(size=8, symbol='square'),
        yaxis='y2'
    )
)

# Configura o layout com os dois eixos Y
fig_combinado.update_layout(
    title=dict(
        text='Evolução Mensal de Pedidos e Receita (Jan/2017 - Ago/2018)',
        x=0.5,
        xanchor='center'
    ),
    xaxis=dict(title='Mês', tickangle=-45),
    yaxis=dict(
        title=dict(text='Quantidade de Pedidos', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53'),
        side='left'
    ),
    yaxis2=dict(
        title=dict(text='Receita Total (R$)', font=dict(color='#9b1c31')),
        tickfont=dict(color='#9b1c31'),
        tickprefix='R$ ',
        tickformat=',.0f',
        side='right',
        overlaying='y'
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_combinado.show()

**Evolução da receita:**

A receita cresceu de R$ 111 mil em janeiro/2017 para R$ 987 mil em novembro/2017. A Black Friday foi o ponto mais alto do período. Em 2018, a receita mensal estabilizou na faixa de R$ 800 mil a R$ 977 mil.

**Ritmo de crescimento da receita:**

A receita teve crescimento semelhante ao dos pedidos: +109% em fevereiro/2017 e +52% em novembro/2017 (Black Friday). A queda de -26% em dezembro/2017 é esperada após o pico de novembro. Em 2018, as variações ficaram em uma faixa estreita, mostrando estabilidade do faturamento mensal.

In [83]:
# Gráfico 3: Evolução do ticket médio
fig_ticket = px.line(
    evolucao_mensal,
    x='ano_mes_str',
    y='ticket_medio',
    title='Evolução Mensal do Ticket Médio (jan/2017 - ago/2018)',
    labels={'ano_mes_str': 'Mês', 'ticket_medio': 'Ticket Médio (R$)'},
    markers=True
)

fig_ticket.show()

**Comportamento do ticket médio:**

Diferente dos pedidos e da receita, o ticket médio se manteve relativamente estável ao longo do período, oscilando entre R$ 124 e R$ 149. Isso indica que o crescimento da Olist veio principalmente do aumento no volume de clientes, não de clientes gastando mais por compra.

## Análise Geográfica

Esta análise explora a distribuição geográfica das vendas para identificar oportunidades de expansão e concentração de risco.

As perguntas que vamos responder:

1. Em quais estados a Olist mais vende?
2. A operação está concentrada no Sudeste ou nacional?
3. Cliente do Sul gasta mais que do Nordeste?

In [84]:
# Primeira olhada na tabela de clientes
df_customers.head()

,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


In [85]:
# Quantos estados temos no total
print(f'Total de estados: {df_customers["customer_state"].nunique()}')
print(f'Total de clientes: {df_customers.shape[0]:,}')
print(f'Clientes únicos: {df_customers["customer_unique_id"].nunique():,}')

Total de estados: 27
Total de clientes: 99,441
Clientes únicos: 96,096


In [86]:
# Top 10 estados com mais clientes
df_customers['customer_state'].value_counts().head(10)

customer_state
SP    41746
RJ    12852
MG    11635
RS     5466
PR     5045
SC     3637
BA     3380
DF     2140
ES     2033
GO     2020
Name: count, dtype: int64

**Primeira observação:** São Paulo concentra 42% dos clientes (41.746 de 99.441), seguido por Rio de Janeiro (13%) e Minas Gerais (12%). Os três estados do Sudeste juntos representam mais de 66% da base de clientes.

### Top 10 Estados em Receita

Saber quantos clientes tem cada estado é uma parte da história. Mais importante é saber **quanto cada estado gera de receita**. Vamos cruzar a `df_vendas_entregues` com a `df_customers` para descobrir.

In [87]:
# Junta vendas entregues com clientes pra ter estado de cada venda
df_vendas_geo = pd.merge(
    df_vendas_entregues,
    df_customers[['customer_id', 'customer_state']],
    on='customer_id',
    how='inner'
)

# Confere o tamanho
print(f'df_vendas_entregues: {df_vendas_entregues.shape[0]:,} linhas')
print(f'df_vendas_geo: {df_vendas_geo.shape[0]:,} linhas')

df_vendas_geo.head()

df_vendas_entregues: 110,197 linhas
df_vendas_geo: 110,197 linhas


,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,ano_mes,customer_state
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29,3ce436f183e68e07877b285a838db11a,delivered,2017-09-13 08:59:02,2017-09-13 09:45:35,2017-09-19 18:34:16,2017-09-20 23:43:48,2017-09-29 00:00:00,2017-09,RJ
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93,f6dd3ec061db4e3987629fe6b26e5cce,delivered,2017-04-26 10:53:06,2017-04-26 11:05:13,2017-05-04 14:35:00,2017-05-12 16:04:24,2017-05-15 00:00:00,2017-04,SP
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87,6489ae5e4333f3693df5ad4372dab6d3,delivered,2018-01-14 14:33:31,2018-01-14 14:48:30,2018-01-16 12:36:48,2018-01-22 13:19:16,2018-02-05 00:00:00,2018-01,MG
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79,d4eb9395c8c0431ee92fce09860c5a06,delivered,2018-08-08 10:00:35,2018-08-08 10:10:18,2018-08-10 13:28:00,2018-08-14 13:32:39,2018-08-20 00:00:00,2018-08,SP
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14,58dbd0b2d70206bf40e62cd34e84d795,delivered,2017-02-04 13:57:51,2017-02-04 14:10:13,2017-02-16 09:46:09,2017-03-01 16:42:31,2017-03-17 00:00:00,2017-02,SP


In [88]:
# Calcula receita, pedidos e ticket médio por estado
receita_por_estado = df_vendas_geo.groupby('customer_state').agg(
    receita_total=('price', 'sum'),
    qtd_pedidos=('order_id', 'nunique')
).reset_index()

# Calcula o ticket médio
receita_por_estado['ticket_medio'] = receita_por_estado['receita_total'] / receita_por_estado['qtd_pedidos']

# Calcula o % de receita
total_receita = receita_por_estado['receita_total'].sum()
receita_por_estado['pct_receita'] = (receita_por_estado['receita_total'] / total_receita * 100).round(2)

# Ordena pela receita total (do maior pro menor)
receita_por_estado = receita_por_estado.sort_values('receita_total', ascending=False).reset_index(drop=True)

# Pega só o top 10
top10_estados = receita_por_estado.head(10).copy()

display(top10_estados)

,customer_state,receita_total,qtd_pedidos,ticket_medio,pct_receita
0,SP,5067633.16,40501,125.123655,38.33
1,RJ,1759651.13,12350,142.481873,13.31
2,MG,1552481.83,11354,136.734352,11.74
3,RS,728897.47,5345,136.369966,5.51
4,PR,666063.51,4923,135.296264,5.04
5,SC,507012.13,3546,142.981424,3.83
6,BA,493584.14,3256,151.592181,3.73
7,DF,296498.41,2080,142.547312,2.24
8,GO,282836.70,1957,144.525652,2.14
9,ES,268643.45,1995,134.658371,2.03


In [89]:
import plotly.graph_objects as go

# Cores em gradiente: top 3 em destaque, resto em tons mais claros
cores = ['#000A53', '#000A53', '#000A53',
         '#3d5a96', '#3d5a96', '#3d5a96',
         '#7ba7d9', '#7ba7d9', '#7ba7d9', '#7ba7d9']

# Cria texto combinado: "R$ X,XM (XX%)"
top10_estados['texto_barra'] = top10_estados.apply(
    lambda row: f"R$ {row['receita_total']/1_000_000:.2f}M ({row['pct_receita']}%)",
    axis=1
)

# Cria o gráfico
fig_estados = go.Figure()

fig_estados.add_trace(
    go.Bar(
        x=top10_estados['receita_total'],
        y=top10_estados['customer_state'],
        orientation='h',
        marker=dict(
            color=cores,
            line=dict(color='#000A53', width=1)
        ),
        text=top10_estados['texto_barra'],
        textposition='outside',
        textfont=dict(size=12, color='#000A53')
    )
)

# Configura o layout no padrão do relatório
fig_estados.update_layout(
    title=dict(
        text='<b>Top 10 Estados em Receita</b>',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(
        title='Receita Total (R$)',
        tickprefix='R$ ',
        tickformat=',.0f',
        showgrid=True,
        gridcolor='#e5e5e5',
        range=[0, 6500000]
    ),
    yaxis=dict(
        title='Estado',
        autorange='reversed',
        tickfont=dict(size=13, color='#000A53')
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600,
    margin=dict(l=80, r=200, t=100, b=60)
)

fig_estados.show()

**Top 10 estados em receita:**

São Paulo lidera com folga: representa **38% da receita total** da Olist, mais que a soma dos três próximos estados juntos (RJ + MG + RS = 30,56%). O Sudeste como um todo (SP + RJ + MG + ES) concentra cerca de **65% da receita**, evidenciando alta dependência geográfica.

Vale destacar que **a Bahia tem o maior ticket médio** entre os top 10 (R$ 151,59), maior até que o de São Paulo (R$ 125,12). Isso sugere que clientes do Nordeste, embora em menor volume, fazem compras de maior valor por pedido — possivelmente porque o frete e o acesso a produtos são mais limitados na região, levando a compras mais "concentradas".

Estados do Nordeste (BA, PE, CE) aparecem em posições discretas no ranking, e nenhum estado do Norte aparece no top 10, sinalizando regiões com **menor penetração** e potencial oportunidade de expansão.

### Distribuição por Região

Distribuição de receita entre Norte, Nordeste, Centro-Oeste, Sudeste e Sul.

In [90]:
# Mapeia cada estado para sua região
regioes = {
    'AC': 'Norte', 'AP': 'Norte', 'AM': 'Norte', 'PA': 'Norte',
    'RO': 'Norte', 'RR': 'Norte', 'TO': 'Norte',
    'AL': 'Nordeste', 'BA': 'Nordeste', 'CE': 'Nordeste', 'MA': 'Nordeste',
    'PB': 'Nordeste', 'PE': 'Nordeste', 'PI': 'Nordeste', 'RN': 'Nordeste', 'SE': 'Nordeste',
    'DF': 'Centro-Oeste', 'GO': 'Centro-Oeste', 'MT': 'Centro-Oeste', 'MS': 'Centro-Oeste',
    'ES': 'Sudeste', 'MG': 'Sudeste', 'RJ': 'Sudeste', 'SP': 'Sudeste',
    'PR': 'Sul', 'RS': 'Sul', 'SC': 'Sul'
}

# Cria a coluna de região a partir do estado
df_vendas_geo['regiao'] = df_vendas_geo['customer_state'].map(regioes)

# Confere se criou certo
df_vendas_geo[['customer_state', 'regiao']].head()

,customer_state,regiao
0,RJ,Sudeste
1,SP,Sudeste
2,MG,Sudeste
3,SP,Sudeste
4,SP,Sudeste


In [91]:
# Calcula receita, pedidos e clientes únicos por região
receita_por_regiao = df_vendas_geo.groupby('regiao').agg(
    receita_total=('price', 'sum'),
    qtd_pedidos=('order_id', 'nunique'),
    qtd_clientes=('customer_id', 'nunique')
).reset_index()

# Calcula o ticket médio
receita_por_regiao['ticket_medio'] = receita_por_regiao['receita_total'] / receita_por_regiao['qtd_pedidos']

# Calcula percentual de receita
total_receita = receita_por_regiao['receita_total'].sum()
receita_por_regiao['pct_receita'] = (receita_por_regiao['receita_total'] / total_receita * 100).round(2)

# Arredonda o ticket médio
receita_por_regiao['ticket_medio'] = receita_por_regiao['ticket_medio'].round(2)

# Ordena pela receita (do maior pro menor)
receita_por_regiao = receita_por_regiao.sort_values('receita_total', ascending=False).reset_index(drop=True)

display(receita_por_regiao)

,regiao,receita_total,qtd_pedidos,qtd_clientes,ticket_medio,pct_receita
0,Sudeste,8648409.57,66200,66200,130.64,65.41
1,Sul,1901973.11,13814,13814,137.68,14.39
2,Nordeste,1497083.78,9044,9044,165.53,11.32
3,Centro-Oeste,846956.70,5624,5624,150.60,6.41
4,Norte,327074.95,1796,1796,182.11,2.47


In [133]:
import plotly.graph_objects as go

# Gráfico de barras: distribuição de receita por região
fig_regiao = go.Figure()

fig_regiao.add_trace(
    go.Bar(
        x=receita_por_regiao['regiao'],
        y=receita_por_regiao['receita_total'],
        marker=dict(color='#000A53'),
        text=receita_por_regiao['pct_receita'],
        texttemplate='%{text}%',
        textposition='outside'
    )
)

# Configura o layout no padrão do relatório
fig_regiao.update_layout(
    title=dict(
        text='<b>Distribuição de Receita por Região</b>',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(title='Região'),
    yaxis=dict(
        title=dict(text='Receita Total (R$)', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53'),
        tickprefix='R$ ',
        tickformat=',.0f'
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_regiao.show()

**Distribuição por região:**

A operação da Olist é altamente concentrada no Sudeste, que responde por **65% da receita** total. Sul aparece em segundo lugar com 14%, seguido por Nordeste (11%), Centro-Oeste (6%) e Norte (apenas 2%).

Em outras palavras: **2 em cada 3 reais faturados pela Olist vêm de SP, RJ, MG ou ES**. As regiões Norte e Nordeste juntas representam pouco mais de 13% do negócio, indicando uma das maiores oportunidades estratégicas de expansão para diversificar o risco geográfico.

### Ticket Médio por Região



In [93]:
# Gráfico de barras: ticket médio por região
fig_ticket_regiao = go.Figure()

fig_ticket_regiao.add_trace(
    go.Bar(
        x=receita_por_regiao['regiao'],
        y=receita_por_regiao['ticket_medio'],
        marker=dict(color='#000A53'),
        text=receita_por_regiao['ticket_medio'].round(0),
        texttemplate='R$ %{text}',
        textposition='outside'
    )
)

# Configura o layout no padrão do relatório
fig_ticket_regiao.update_layout(
    title=dict(
        text='<b>Ticket Médio por Região</b>',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(title='Região'),
    yaxis=dict(
        title=dict(text='Ticket Médio (R$)', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53'),
        tickprefix='R$ ',
        tickformat=',.0f'
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_ticket_regiao.show()

**Ticket médio por região:**

Um padrão interessante aparece quando olhamos o ticket médio: regiões com **menor volume** apresentam **ticket médio maior**. O Norte, que representa apenas 2,47% da receita total, tem o maior ticket médio (R$ 182,11), enquanto o Sudeste, com 65% da receita, tem o menor ticket (R$ 130,64) — uma diferença de quase 40%.

Esse comportamento sugere dois perfis de consumo distintos:

- **Sudeste:** muitos clientes comprando com frequência, mas em valores menores (mercado maduro, acesso fácil a produtos)
- **Norte/Nordeste:** poucos clientes, porém com compras de maior valor por pedido (acesso limitado leva a compras mais "concentradas")

Para estratégias de CRM, essa distinção é fundamental: o Sudeste pede campanhas de **frequência** e o Norte/Nordeste, campanhas de **expansão de base** com tickets já naturalmente maiores.

### Conclusão da Análise Geográfica

A análise revelou três descobertas geográficas importantes:

1. **Alta concentração no Sudeste:** quatro estados (SP, RJ, MG, ES) respondem por 65% da receita da Olist, sendo que São Paulo sozinho representa 38%. Isso configura um **risco de concentração geográfica** relevante para o negócio.

2. **Norte e Nordeste são oportunidades inexploradas:** essas duas regiões juntas representam apenas pouco mais de 13% da receita, apesar de concentrarem cerca de 35% da população brasileira. Há aqui um claro **gap de penetração** que pode ser explorado por estratégias de expansão regional.

3. **Ticket médio comporta-se inversamente ao volume:** regiões com menor volume apresentam ticket médio maior (Norte: R$ 182 vs Sudeste: R$ 131). Isso indica que clientes do Norte e Nordeste já chegam com **maior propensão a gastar por pedido** — só falta volume e capilaridade.

**Implicação estratégica:** diversificar geograficamente a base de clientes da Olist é uma algo importante para o crescimento sustentável e de redução de risco. Investir em logística e parcerias regionais para Nordeste e Norte pode acelerar o crescimento explorando clientes que já demonstram alto valor por compra.

---

---

## Mix de Produtos

Quais categorias de produto sustentam o faturamento da Olist. 
Esta análise identifica as categorias mais relevantes e o grau de concentração do mix de produtos.

In [94]:
# Primeira olhada na tabela de produtos
df_products.head()

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0


In [95]:
# Quantas categorias diferentes existem
print(f'Total de categorias diferentes: {df_products["product_category_name"].nunique()}')
print()
# Top 10 categorias com mais produtos cadastrados
df_products['product_category_name'].value_counts().head(10)

Total de categorias diferentes: 73



product_category_name
cama_mesa_banho           3029
esporte_lazer             2867
moveis_decoracao          2657
beleza_saude              2444
utilidades_domesticas     2335
automotivo                1900
informatica_acessorios    1639
brinquedos                1411
relogios_presentes        1329
telefonia                 1134
Name: count, dtype: int64

In [96]:
# Junta items com produtos pra ter a categoria de cada venda
df_vendas_categoria = pd.merge(
    df_vendas_entregues,
    df_products[['product_id', 'product_category_name']],
    on='product_id',
    how='inner'
)

# Confere o tamanho
print(f'df_vendas_entregues: {df_vendas_entregues.shape[0]:,} linhas')
print(f'df_vendas_categoria: {df_vendas_categoria.shape[0]:,} linhas')

df_vendas_categoria[['order_id', 'product_id', 'product_category_name', 'price']].head()

df_vendas_entregues: 110,197 linhas
df_vendas_categoria: 110,197 linhas


,order_id,product_id,product_category_name,price
0,00010242fe8c5a6d1ba2dd792cb16214,4244733e06e7ecb4970a6e2683c13e61,cool_stuff,58.90
1,00018f77f2f0320c557190d7a144bdd3,e5f2d52b802189ee658865ca93d83a8f,pet_shop,239.90
2,000229ec398224ef6ca0657da4fc703e,c777355d18b72b67abbeef9df44fd0fd,moveis_decoracao,199.00
3,00024acbcdf0a6daa1e931b038114c75,7634da152a4610f1595efa32f14722fc,perfumaria,12.99
4,00042b26cf59d7ce69dfabb4e55b4fd9,ac6c3623068f30de03045865e4e10089,ferramentas_jardim,199.90


In [97]:
# Calcula receita, pedidos e ticket médio por categoria
receita_por_categoria = df_vendas_categoria.groupby('product_category_name').agg(
    receita_total=('price', 'sum'),
    qtd_itens=('price', 'count')
).reset_index()

# Dicionário de tradução das categorias
traducao_categorias = {
    'beleza_saude': 'Beleza & Saúde',
    'relogios_presentes': 'Relógios & Presentes',
    'cama_mesa_banho': 'Cama, Mesa & Banho',
    'esporte_lazer': 'Esporte & Lazer',
    'informatica_acessorios': 'Informática & Acessórios',
    'moveis_decoracao': 'Móveis & Decoração',
    'utilidades_domesticas': 'Utilidades Domésticas',
    'cool_stuff': 'Cool Stuff',
    'automotivo': 'Automotivo',
    'brinquedos': 'Brinquedos',
    'ferramentas_jardim': 'Ferramentas & Jardim',
    'perfumaria': 'Perfumaria',
    'pet_shop': 'Pet Shop',
    'telefonia': 'Telefonia',
    'papelaria': 'Papelaria',
    'eletrodomesticos': 'Eletrodomésticos',
    'fashion_bolsas_e_acessorios': 'Moda — Bolsas & Acessórios',
    'eletronicos': 'Eletrônicos',
    'bebes': 'Bebês',
    'consoles_games': 'Consoles & Games'
}

# Aplica a tradução (categorias não mapeadas mantêm o nome original)
receita_por_categoria['product_category_name'] = receita_por_categoria['product_category_name'].replace(traducao_categorias)

# Calcula o ticket médio por item
receita_por_categoria['ticket_medio'] = receita_por_categoria['receita_total'] / receita_por_categoria['qtd_itens']

# Calcula percentual de receita
total_receita = receita_por_categoria['receita_total'].sum()
receita_por_categoria['pct_receita'] = (receita_por_categoria['receita_total'] / total_receita * 100).round(2)

# Arredonda o ticket médio
receita_por_categoria['ticket_medio'] = receita_por_categoria['ticket_medio'].round(2)

# Ordena pela receita (do maior pro menor)
receita_por_categoria = receita_por_categoria.sort_values('receita_total', ascending=False).reset_index(drop=True)

# Calcula percentual acumulado (Pareto)
receita_por_categoria['pct_acumulado'] = receita_por_categoria['pct_receita'].cumsum().round(2)

# Mostra o top 10
top10_categorias = receita_por_categoria.head(10).copy()

display(top10_categorias)

,product_category_name,receita_total,qtd_itens,ticket_medio,pct_receita,pct_acumulado
0,Beleza & Saúde,1233131.72,9465,130.28,9.45,9.45
1,Relógios & Presentes,1166176.98,5859,199.04,8.94,18.39
2,"Cama, Mesa & Banho",1023434.76,10953,93.44,7.84,26.23
3,Esporte & Lazer,954852.55,8431,113.25,7.32,33.55
4,Informática & Acessórios,888724.61,7644,116.26,6.81,40.36
5,Móveis & Decoração,711927.69,8160,87.25,5.46,45.82
6,Utilidades Domésticas,615628.69,6795,90.60,4.72,50.54
7,Cool Stuff,610204.10,3718,164.12,4.68,55.22
8,Automotivo,578966.65,4140,139.85,4.44,59.66
9,Brinquedos,471286.48,4030,116.94,3.61,63.27


In [98]:
import plotly.graph_objects as go

# Gráfico de barras horizontal: top 10 categorias em receita
fig_categorias = go.Figure()

fig_categorias.add_trace(
    go.Bar(
        x=top10_categorias['receita_total'],
        y=top10_categorias['product_category_name'],
        orientation='h',
        marker=dict(color='#000A53'),
        text=top10_categorias['pct_receita'],
        texttemplate='%{text}%',
        textposition='outside'
    )
)

fig_categorias.update_layout(
    title=dict(
        text='Top 10 Categorias em Receita',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(
        title='Receita Total (R$)',
        tickprefix='R$ ',
        tickformat=',.0f',
        range=[0, 1400000]
    ),
    yaxis=dict(
        title='Categoria',
        autorange='reversed'
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_categorias.show()

**O que sustenta a receita da Olist**

A receita da Olist se distribui entre 73 categorias de produto, mas a concentração é expressiva: as dez maiores respondem por 63% de todo o faturamento. Beleza & Saúde lidera com R$ 1,23 milhão (9,5% do total), seguida de perto por Relógios & Presentes, com R$ 1,17 milhão.

O contraste entre volume e valor é o ponto mais revelador. Cama, Mesa & Banho vendeu o maior número de itens de todo o catálogo (10.953 unidades), mas fica apenas em terceiro lugar em receita — seu ticket médio de R$ 93 é o menor do top 10. Já Relógios & Presentes chega ao segundo lugar vendendo quase metade disso (5.859 itens), sustentada por um ticket médio de R$ 199, mais que o dobro. Em outras palavras: há categorias que faturam por volume e categorias que faturam por valor agregado.

Cool Stuff é outro exemplo de eficiência por valor: com apenas 3.718 itens vendidos — um terço do volume de Cama, Mesa & Banho — gera receita comparável, graças ao ticket de R$ 164.

Essa leitura orienta a estratégia comercial. Categorias de ticket alto e bom volume (Relógios & Presentes, Cool Stuff) são candidatas naturais a ações de destaque e cross-sell. Já categorias de alto volume e ticket baixo (Cama, Mesa & Banho, Móveis & Decoração) pedem estratégias de aumento de valor por pedido, como kits e combos.

# Comportamento e Pagamentos

In [99]:
# Primeiras linhas da tabela de pagamentos
df_payments.head()

,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


In [100]:
# Informações da tabela
df_payments.info()

<class 'pandas.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  str    
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  str    
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), str(2)
memory usage: 4.0 MB


In [101]:
# Conta quantos pedidos usaram cada forma de pagamento
formas_pagamento = df_payments['payment_type'].value_counts().reset_index()
formas_pagamento.columns = ['forma_pagamento', 'quantidade']

# Calcula o percentual
total = formas_pagamento['quantidade'].sum()
formas_pagamento['percentual'] = (formas_pagamento['quantidade'] / total * 100).round(2)

# Traduz os nomes pra português
formas_pagamento['forma_pagamento'] = formas_pagamento['forma_pagamento'].replace({
    'credit_card': 'Cartão de Crédito',
    'boleto': 'Boleto',
    'voucher': 'Voucher',
    'debit_card': 'Cartão de Débito',
    'not_defined': 'Não Definido'
})

display(formas_pagamento)



,forma_pagamento,quantidade,percentual
0,Cartão de Crédito,76795,73.92
1,Boleto,19784,19.04
2,Voucher,5775,5.56
3,Cartão de Débito,1529,1.47
4,Não Definido,3,0.00


In [102]:
import plotly.express as px

# Gráfico de pizza dos meios de pagamento com paleta monocromática
fig_pagamento = px.pie(
    formas_pagamento,
    values='quantidade',
    names='forma_pagamento',
    title='Distribuição dos Meios de Pagamento',
    hole=0.4,
    color_discrete_sequence=['#1f3a5f', '#3d6ea5', '#7ba7d9', '#b8d4f0']
)

fig_pagamento.update_traces(
    textposition='outside',
    textinfo='percent+label'
)

fig_pagamento.update_layout(
    showlegend=False,
    width=700,
    height=500
)

fig_pagamento.show()

In [103]:
# Calcula o valor médio e total por forma de pagamento
valor_por_pagamento = df_payments.groupby('payment_type').agg(
    valor_medio=('payment_value', 'mean'),
    valor_total=('payment_value', 'sum')
).reset_index()

# Traduz os nomes
valor_por_pagamento['payment_type'] = valor_por_pagamento['payment_type'].replace({
    'credit_card': 'Cartão de Crédito',
    'boleto': 'Boleto',
    'voucher': 'Voucher',
    'debit_card': 'Cartão de Débito',
    'not_defined': 'Não Definido'
})

# Renomeia a coluna pra ficar mais bonita
valor_por_pagamento = valor_por_pagamento.rename(columns={'payment_type': 'forma_pagamento'})

# Ordena pelo valor médio (do maior pro menor)
valor_por_pagamento = valor_por_pagamento.sort_values('valor_medio', ascending=False).reset_index(drop=True)

display(valor_por_pagamento)

,forma_pagamento,valor_medio,valor_total
0,Cartão de Crédito,163.319021,12542084.19
1,Boleto,145.034435,2869361.27
2,Cartão de Débito,142.570170,217989.79
3,Voucher,65.703354,379436.87
4,Não Definido,0.000000,0.00


In [104]:
# Filtra só cartão de crédito (única forma com parcelamento real)
df_credito = df_payments[df_payments['payment_type'] == 'credit_card']

# Conta quantos pedidos foram pagos em cada número de parcelas
parcelas = df_credito['payment_installments'].value_counts().reset_index()
parcelas.columns = ['num_parcelas', 'quantidade']

# Calcula o percentual
total_credito = parcelas['quantidade'].sum()
parcelas['percentual'] = (parcelas['quantidade'] / total_credito * 100).round(2)

# Ordena pelo número de parcelas (1, 2, 3, ...)
parcelas = parcelas.sort_values('num_parcelas').reset_index(drop=True)

display(parcelas)

,num_parcelas,quantidade,percentual
0,0,2,0.00
1,1,25455,33.15
2,2,12413,16.16
3,3,10461,13.62
4,4,7098,9.24
5,5,5239,6.82
6,6,3920,5.10
7,7,1626,2.12
8,8,4268,5.56
9,9,644,0.84


In [105]:
# Gráfico de barras: distribuição de parcelas
fig_parcelas = px.bar(
    parcelas,
    x='num_parcelas',
    y='quantidade',
    title='Distribuição de Parcelas no Cartão de Crédito',
    labels={'num_parcelas': 'Número de Parcelas', 'quantidade': 'Quantidade de Pagamentos'},
    text='percentual'
)

# Adiciona o símbolo % nas barras
fig_parcelas.update_traces(texttemplate='%{text}%', textposition='outside')

fig_parcelas.show()

In [106]:
# Função pra agrupar parcelas em faixas
def agrupar_parcelas(num):
    if num == 1:
        return '1. À vista'
    elif num <= 4:
        return '2. Curto (2-4x)'
    elif num <= 10:
        return '3. Médio (5-10x)'
    else:
        return '4. Longo (acima de 10x)'

# Aplica a função na coluna de parcelas
df_credito['faixa_parcela'] = df_credito['payment_installments'].apply(agrupar_parcelas)

# Conta quantos pagamentos em cada faixa
faixas = df_credito['faixa_parcela'].value_counts().reset_index()
faixas.columns = ['faixa', 'quantidade']

# Calcula o percentual
total_credito = faixas['quantidade'].sum()
faixas['percentual'] = (faixas['quantidade'] / total_credito * 100).round(2)

# Ordena alfabeticamente (por isso usei números no nome da faixa)
faixas = faixas.sort_values('faixa').reset_index(drop=True)

display(faixas)

,faixa,quantidade,percentual
0,1. À vista,25455,33.15
1,2. Curto (2-4x),29974,39.03
2,3. Médio (5-10x),21025,27.38
3,4. Longo (acima de 10x),341,0.44


In [107]:
# Calcula o ticket médio por faixa de parcelas
ticket_por_faixa = df_credito.groupby('faixa_parcela').agg(
    ticket_medio=('payment_value', 'mean'),
    quantidade=('payment_value', 'count')
).reset_index()

# Renomeia a coluna pra ficar mais clara
ticket_por_faixa = ticket_por_faixa.rename(columns={'faixa_parcela': 'faixa'})

# Arredonda o ticket médio
ticket_por_faixa['ticket_medio'] = ticket_por_faixa['ticket_medio'].round(2)

# Ordena pelas faixas (1, 2, 3, 4)
ticket_por_faixa = ticket_por_faixa.sort_values('faixa').reset_index(drop=True)

display(ticket_por_faixa)

,faixa,ticket_medio,quantidade
0,1. À vista,95.87,25455
1,2. Curto (2-4x),141.27,29974
2,3. Médio (5-10x),273.24,21025
3,4. Longo (acima de 10x),358.34,341


In [108]:
import plotly.graph_objects as go

# Cria a figura vazia
fig_ticket_faixa = go.Figure()

# Cores em gradiente: do mais claro (à vista) ao mais escuro (parcelamento longo)
cores_ticket = ['#7ba7d9', '#3d5a96', '#1a3d6b', '#000A53']

# Cria texto formatado em negrito
ticket_por_faixa['texto_barra'] = ticket_por_faixa['ticket_medio'].apply(
    lambda v: f'<b>R$ {round(v)}</b>'
)

# Adiciona as barras com o ticket médio
fig_ticket_faixa.add_trace(
    go.Bar(
        x=ticket_por_faixa['faixa'],
        y=ticket_por_faixa['ticket_medio'],
        marker=dict(
            color=cores_ticket,
            line=dict(color='#000A53', width=1)
        ),
        text=ticket_por_faixa['texto_barra'],
        textposition='outside',
        textfont=dict(size=14, color='#000A53')
    )
)

# Configura o layout no padrão do relatório
fig_ticket_faixa.update_layout(
    title=dict(
        text='<b>Ticket Médio por Faixa de Parcelamento</b>',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(title='Faixa de Parcelas'),
    yaxis=dict(
        title=dict(text='Ticket Médio (R$)', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53'),
        tickprefix='R$ ',
        tickformat=',.0f',
        range=[0, 520]
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_ticket_faixa.show()

**Ticket médio por faixa de parcelamento:**

Há uma relação clara entre o número de parcelas e o valor da compra: quanto mais parcelas, maior o ticket. Quem paga à vista tem ticket médio de R$ 96, enquanto quem parcela de forma mais longa gasta progressivamente mais — chegando a R$ 273 na faixa de 5 a 10 parcelas e R$ 358 acima de 10 vezes. O parcelamento longo está associado a compras de maior valor, um indício de que os clientes usam essa modalidade para diluir compras maiores.

In [109]:
# Valor médio por número de parcelas
valor_por_parcela = df_credito.groupby('payment_installments').agg(
    valor_medio=('payment_value', 'mean'),
    quantidade=('payment_value', 'count')
).reset_index()

# Renomeia a coluna pra ficar mais clara
valor_por_parcela = valor_por_parcela.rename(columns={'payment_installments': 'num_parcelas'})

display(valor_por_parcela)

,num_parcelas,valor_medio,quantidade
0,0,94.315000,2
1,1,95.872930,25455
2,2,127.228150,12413
3,3,142.539317,10461
4,4,163.976840,7098
5,5,183.465222,5239
6,6,209.849952,3920
7,7,187.673672,1626
8,8,307.737427,4268
9,9,203.440870,644


In [110]:
# Remove parcelas com 0 (erro) e com menos de 30 pagamentos (amostra pequena)
valor_por_parcela_limpo = valor_por_parcela[
    (valor_por_parcela['num_parcelas'] > 0) & 
    (valor_por_parcela['quantidade'] >= 30)
].reset_index(drop=True)

display(valor_por_parcela_limpo)

,num_parcelas,valor_medio,quantidade
0,1,95.872930,25455
1,2,127.228150,12413
2,3,142.539317,10461
3,4,163.976840,7098
4,5,183.465222,5239
5,6,209.849952,3920
6,7,187.673672,1626
7,8,307.737427,4268
8,9,203.440870,644
9,10,415.085837,5328


In [111]:
import plotly.graph_objects as go

fig_relacao = go.Figure()

# Cria texto com formatação "R$ X" arredondado corretamente
valor_por_parcela_limpo['texto_bolinha'] = valor_por_parcela_limpo['valor_medio'].apply(
    lambda v: f'R$ {round(v)}'
)

# Cria texto customizado pro tooltip
valor_por_parcela_limpo['texto_tooltip'] = valor_por_parcela_limpo.apply(
    lambda row: (
        f"<b>{int(row['num_parcelas'])}x</b><br>"
        f"Valor médio: R$ {round(row['valor_medio'])}<br>"
        f"Pagamentos: {int(row['quantidade']):,}"
    ),
    axis=1
)

fig_relacao.add_trace(
    go.Scatter(
        x=valor_por_parcela_limpo['num_parcelas'],
        y=valor_por_parcela_limpo['valor_medio'],
        mode='markers+text',
        marker=dict(
            size=valor_por_parcela_limpo['quantidade'],
            sizemode='area',
            sizeref=2.*max(valor_por_parcela_limpo['quantidade'])/(60.**2),
            sizemin=8,
            color='#000A53',
            opacity=0.7,
            line=dict(color='#000A53', width=2)
        ),
        text=valor_por_parcela_limpo['texto_bolinha'],
        textposition='top center',
        textfont=dict(size=11, color='#000A53'),
        hovertext=valor_por_parcela_limpo['texto_tooltip'],
        hoverinfo='text'
    )
)

# Layout
fig_relacao.update_layout(
    title=dict(
        text='<b>Valor Médio por Número de Parcelas</b>',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(
        title='Número de Parcelas',
        dtick=1,
        showgrid=True,
        gridcolor='#e5e5e5'
    ),
    yaxis=dict(
        title=dict(text='Valor Médio (R$)', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53'),
        tickprefix='R$ ',
        tickformat=',.0f',
        showgrid=True,
        gridcolor='#e5e5e5'
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600,
    margin=dict(l=80, r=80, t=120, b=60)
)

fig_relacao.show()

**Relação entre parcelas e ticket médio:**

Há uma relação clara entre o número de parcelas e o valor médio do pagamento: quanto maior o parcelamento, maior o valor da compra. Pagamentos à vista (1x) concentram o maior volume (~25 mil pagamentos) com ticket de R$ 96 — representado pela bolinha gigante no canto inferior esquerdo. À medida que o número de parcelas aumenta, o ticket cresce, chegando a R$ 415 em 10x e R$ 446 em 15x — visíveis nas bolinhas menores no canto superior direito.

Isso indica dois perfis de cliente:

- **Cliente de tickets baixos:** prefere pagar à vista ou parcelar pouco. Representa o maior volume da operação.
- **Cliente de tickets altos:** dilui o pagamento em mais parcelas. É minoria em volume, mas concentra ticket médio elevado.

Para estratégias de CRM e segmentação, essa distinção é fundamental — cada perfil exige comunicação, oferta e canal diferentes.

### Conclusão: Comportamento e Pagamentos

A análise dos pagamentos revelou três comportamentos principais:

1. **Cartão de crédito domina:** 74% das transações usam cartão de crédito, seguido pelo boleto (19%). Além do volume, o cartão também tem o maior ticket médio (~R$ 163), enquanto voucher e débito têm participação marginal tanto em volume quanto em valor.

2. **Brasileiro prefere pagar à vista ou em poucas parcelas:** cerca de metade dos pagamentos em cartão é feita em 1x, e mais de 70% se concentra entre 1 e 4 parcelas. Parcelamentos longos (acima de 10x) representam menos de 1% das transações.

3. **Quanto maior o parcelamento, maior o ticket:** quem paga à vista tem ticket médio de R$ 96, enquanto pagamentos em 10x chegam a R$ 415 — mais de 4x a diferença. Olhando por faixas agrupadas, a progressão é gradual: à vista (R$ 96), curto/2-4x (R$ 141), médio/5-10x (R$ 273) e longo/acima de 10x (R$ 358).

Isso desenha dois perfis de cliente: o de alto volume e ticket baixo, que prefere pagar à vista, e o de baixo volume e ticket alto, que dilui compras maiores em mais parcelas. Para estratégias de CRM, essa distinção é fundamental — cada perfil exige oferta, canal e comunicação diferentes.

# Satisfação do Cliente

In [112]:
# Conta quantas reviews existem em cada nota (1 a 5)
notas = df_reviews['review_score'].value_counts().reset_index()
notas.columns = ['nota', 'quantidade']

# Calcula o percentual
total_reviews = notas['quantidade'].sum()
notas['percentual'] = (notas['quantidade'] / total_reviews * 100).round(2)

# Ordena da nota 1 até a 5
notas = notas.sort_values('nota').reset_index(drop=True)

display(notas)

,nota,quantidade,percentual
0,1,11424,11.51
1,2,3151,3.18
2,3,8179,8.24
3,4,19142,19.29
4,5,57328,57.78


In [113]:
# Cria o gráfico de barras com a distribuição das notas
fig_notas = px.bar(
    notas,
    x='nota',
    y='percentual',
    title='Distribuição das Notas de Avaliação',
    labels={'nota': 'Nota (1 a 5)', 'percentual': 'Percentual (%)'},
    text='percentual',
    template='plotly_white'
)

# Adiciona o símbolo % no número que aparece em cima de cada barra
fig_notas.update_traces(texttemplate='%{text}%', textposition='outside')

# Aumenta a escala do eixo Y para os textos não ficarem apertados no topo
fig_notas.update_yaxes(range=[0, 65])

fig_notas.show()

**Distribuição das notas:**

A maioria dos clientes está satisfeita: 77% dão notas 4 ou 5. Por outro lado, 11,5% dão nota mínima (1), enquanto notas 2 e 3 são bem menos frequentes. Isso mostra um padrão típico de e-commerce: clientes muito satisfeitos ou muito insatisfeitos são os que mais avaliam, enquanto experiências "meio termo" tendem a não gerar review.

In [114]:
# Converte a coluna de data pra tipo datetime
df_reviews['review_creation_date'] = pd.to_datetime(df_reviews['review_creation_date'])

# Cria coluna nova com ano-mês
df_reviews['ano_mes'] = df_reviews['review_creation_date'].dt.to_period('M')

# Confere se ficou certo
df_reviews[['review_creation_date', 'ano_mes', 'review_score']].head()

,review_creation_date,ano_mes,review_score
0,2018-01-18,2018-01,4
1,2018-03-10,2018-03,5
2,2018-02-17,2018-02,5
3,2017-04-21,2017-04,5
4,2018-03-01,2018-03,5


In [115]:
# Agrupa por mês e calcula a nota média e quantidade de reviews
evolucao_satisfacao = df_reviews.groupby('ano_mes').agg(
    nota_media=('review_score', 'mean'),
    qtd_reviews=('review_score', 'count')
).reset_index()

# Ordena por mês (do mais antigo pro mais novo)
evolucao_satisfacao = evolucao_satisfacao.sort_values('ano_mes').reset_index(drop=True)

# Mostra o resultado
display(evolucao_satisfacao)

,ano_mes,nota_media,qtd_reviews
0,2016-10,4.055866,179
1,2016-11,3.188119,101
2,2016-12,2.355556,45
3,2017-01,4.338912,239
4,2017-02,4.280962,1413
5,2017-03,4.033051,2481
6,2017-04,4.036983,2055
7,2017-05,4.100539,3710
8,2017-06,4.127616,3440
9,2017-07,4.183271,3503


In [116]:
# Filtra só a partir de janeiro/2017
satisfacao_mensal = evolucao_satisfacao[evolucao_satisfacao['ano_mes'] >= '2017-01'].reset_index(drop=True)

# Converte ano_mes pra texto (pra usar no gráfico)
satisfacao_mensal['ano_mes_str'] = satisfacao_mensal['ano_mes'].astype(str)

display(satisfacao_mensal)

,ano_mes,nota_media,qtd_reviews,ano_mes_str
0,2017-01,4.338912,239,2017-01
1,2017-02,4.280962,1413,2017-02
2,2017-03,4.033051,2481,2017-03
3,2017-04,4.036983,2055,2017-04
4,2017-05,4.100539,3710,2017-05
5,2017-06,4.127616,3440,2017-06
6,2017-07,4.183271,3503,2017-07
7,2017-08,4.224961,4503,2017-08
8,2017-09,4.182424,4199,2017-09
9,2017-10,4.182188,4424,2017-10


In [117]:
# Gráfico de linha mostrando a evolução da nota média ao longo dos meses
fig_satisfacao = px.line(
    satisfacao_mensal,
    x='ano_mes_str',
    y='nota_media',
    title='Evolução Mensal da Nota Média (jan/2017 - ago/2018)',
    labels={'ano_mes_str': 'Mês', 'nota_media': 'Nota Média (1 a 5)'},
    markers=True,
    template='plotly_white'
)

# Ajusta o eixo Y para uma escala que mostre bem as variações
fig_satisfacao.update_yaxes(range=[3.5, 4.5])

fig_satisfacao.show()

**Evolução da satisfação:**

A nota média ficou geralmente entre 4,0 e 4,3 ao longo do período, indicando uma satisfação consistente. Porém, dois momentos chamam atenção: dezembro/2017 (3,93) logo após a Black Friday, e março/2018 (3,72), que foi a pior nota do período analisado. Em ambos os casos, é possível que o aumento de volume tenha pressionado a operação, afetando a qualidade da entrega. A partir de maio/2018, a nota se recuperou, voltando ao patamar de 4,2.

In [118]:
# Junta pedidos com reviews pra ter nota e datas no mesmo lugar
df_pedidos_reviews = pd.merge(
    df_orders,
    df_reviews[['order_id', 'review_score']],
    on='order_id',
    how='inner'
)

# Confere o tamanho
print(f'df_orders: {df_orders.shape[0]:,} pedidos')
print(f'df_reviews: {df_reviews.shape[0]:,} reviews')
print(f'df_pedidos_reviews: {df_pedidos_reviews.shape[0]:,} linhas')

df_pedidos_reviews.head()

df_orders: 99,441 pedidos
df_reviews: 99,224 reviews
df_pedidos_reviews: 99,224 linhas


,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,review_score
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00,4
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00,4
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00,5
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00,5
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00,5


In [119]:
# Filtra só pedidos entregues (que têm data de entrega real)
df_entregues = df_pedidos_reviews[df_pedidos_reviews['order_status'] == 'delivered'].copy()

# Converte as datas pra tipo datetime
df_entregues['order_delivered_customer_date'] = pd.to_datetime(df_entregues['order_delivered_customer_date'])
df_entregues['order_estimated_delivery_date'] = pd.to_datetime(df_entregues['order_estimated_delivery_date'])

# Calcula a diferença em dias entre entrega real e estimada
# Negativo = adiantado / Positivo = atrasado
df_entregues['dias_atraso'] = (df_entregues['order_delivered_customer_date'] - df_entregues['order_estimated_delivery_date']).dt.days

# Confere o tamanho
print(f'Pedidos entregues com review: {df_entregues.shape[0]:,}')

# Mostra as primeiras linhas pra conferir
df_entregues[['order_id', 'order_estimated_delivery_date', 'order_delivered_customer_date', 'dias_atraso', 'review_score']].head()

Pedidos entregues com review: 96,361


,order_id,order_estimated_delivery_date,order_delivered_customer_date,dias_atraso,review_score
0,e481f51cbdc54678b7cc49136f2d6af7,2017-10-18,2017-10-10 21:25:13,-8.0,4
1,53cdb2fc8bc7dce0b6741e2150273451,2018-08-13,2018-08-07 15:27:45,-6.0,4
2,47770eb9100c2d0c44946d9cf07ec65d,2018-09-04,2018-08-17 18:06:29,-18.0,5
3,949d5b44dbf5de918fe9c16f97b45f8a,2017-12-15,2017-12-02 00:28:42,-13.0,5
4,ad21c59c0840e6cb83a9ceb5573f8159,2018-02-26,2018-02-16 18:17:02,-10.0,5


In [120]:
# Função pra agrupar em faixas de atraso
def faixa_atraso(dias):
    if dias <= -10:
        return '1. Muito adiantado (10+ dias antes)'
    elif dias <= -1:
        return '2. Adiantado (1-10 dias antes)'
    elif dias == 0:
        return '3. No prazo'
    elif dias <= 10:
        return '4. Atrasado (1-10 dias depois)'
    else:
        return '5. Muito atrasado (10+ dias depois)'

# Aplica a função na coluna de dias_atraso
df_entregues['faixa_entrega'] = df_entregues['dias_atraso'].apply(faixa_atraso)

# Calcula a nota média e quantidade por faixa
nota_por_faixa = df_entregues.groupby('faixa_entrega').agg(
    nota_media=('review_score', 'mean'),
    quantidade=('review_score', 'count')
).reset_index()

# Calcula percentual
total = nota_por_faixa['quantidade'].sum()
nota_por_faixa['percentual'] = (nota_por_faixa['quantidade'] / total * 100).round(2)

# Ordena pelas faixas (1, 2, 3, 4, 5)
nota_por_faixa = nota_por_faixa.sort_values('faixa_entrega').reset_index(drop=True)

# Arredonda a nota média pra 2 casas
nota_por_faixa['nota_media'] = nota_por_faixa['nota_media'].round(2)

display(nota_por_faixa)

,faixa_entrega,nota_media,quantidade,percentual
0,1. Muito adiantado (10+ dias antes),4.32,61884,64.22
1,2. Adiantado (1-10 dias antes),4.23,26769,27.78
2,3. No prazo,4.03,1291,1.34
3,4. Atrasado (1-10 dias depois),2.53,4371,4.54
4,5. Muito atrasado (10+ dias depois),1.72,2046,2.12


In [121]:
# Vamos ver a % de pedidos atrasados por mês
df_entregues['ano_mes_entrega'] = pd.to_datetime(df_entregues['order_purchase_timestamp']).dt.to_period('M')

# Cria coluna que marca se foi atraso (dias > 0)
df_entregues['atrasou'] = df_entregues['dias_atraso'] > 0

# Calcula % de atraso por mês
atraso_por_mes = df_entregues.groupby('ano_mes_entrega').agg(
    total_pedidos=('order_id', 'count'),
    pedidos_atrasados=('atrasou', 'sum')
).reset_index()

atraso_por_mes['pct_atraso'] = (atraso_por_mes['pedidos_atrasados'] / atraso_por_mes['total_pedidos'] * 100).round(2)

# Filtra a partir de jan/2017 e mostra
atraso_por_mes = atraso_por_mes[atraso_por_mes['ano_mes_entrega'] >= '2017-01'].reset_index(drop=True)

display(atraso_por_mes)

,ano_mes_entrega,total_pedidos,pedidos_atrasados,pct_atraso
0,2017-01,748,21,2.81
1,2017-02,1651,42,2.54
2,2017-03,2542,113,4.45
3,2017-04,2296,148,6.45
4,2017-05,3554,99,2.79
5,2017-06,3142,92,2.93
6,2017-07,3882,107,2.76
7,2017-08,4207,123,2.92
8,2017-09,4144,177,4.27
9,2017-10,4474,179,4.00


In [134]:
import plotly.graph_objects as go

# Cria a figura vazia
fig_atraso_nota = go.Figure()

# Cores em gradiente: verde-azulado (bom) escurecendo até o vermelho-escuro (ruim)
# Como é satisfação caindo, uso azul-marinho que escurece e a última vermelha (alerta)
cores_atraso = ['#1a3d6b', '#3d5a96', '#7ba7d9', '#c0392b', '#9b1c31']

# Texto formatado em negrito
nota_por_faixa['texto_barra'] = nota_por_faixa['nota_media'].apply(
    lambda v: f'<b>{v:.2f}</b>'
)

# Adiciona as barras
fig_atraso_nota.add_trace(
    go.Bar(
        x=nota_por_faixa['faixa_entrega'],
        y=nota_por_faixa['nota_media'],
        marker=dict(
            color=cores_atraso,
            line=dict(color='#000A53', width=1)
        ),
        text=nota_por_faixa['texto_barra'],
        textposition='outside',
        textfont=dict(size=14, color='#000A53')
    )
)

# Configura o layout no padrão do relatório
fig_atraso_nota.update_layout(
    title=dict(
        text='<b>Nota Média por Faixa de Entrega</b>',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(title='Faixa de Entrega'),
    yaxis=dict(
        title=dict(text='Nota Média (1 a 5)', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53'),
        range=[0, 5.5]
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_atraso_nota.show()

**Relação entre atraso e nota:**

A análise revela uma relação direta e dramática entre o tempo de entrega e a satisfação do cliente:

- Pedidos entregues no prazo ou adiantados recebem notas médias acima de 4 estrelas
- A nota cai 37% quando há qualquer atraso (de 4,03 para 2,53)
- Pedidos com mais de 10 dias de atraso recebem nota média de apenas 1,72

A boa notícia é que 92% dos pedidos chegam antes do prazo prometido, indicando que a Olist trabalha com prazos conservadores e geralmente supera as expectativas. Porém, quando o atraso acontece, o impacto na satisfação é severo.

In [123]:
# Converte a coluna de resposta pra tipo datetime
df_reviews['review_answer_timestamp'] = pd.to_datetime(df_reviews['review_answer_timestamp'])

# Calcula a diferença em dias entre solicitação e resposta
df_reviews['dias_resposta'] = (df_reviews['review_answer_timestamp'] - df_reviews['review_creation_date']).dt.days

# Confere as primeiras linhas
df_reviews[['review_creation_date', 'review_answer_timestamp', 'dias_resposta', 'review_score']].head()

,review_creation_date,review_answer_timestamp,dias_resposta,review_score
0,2018-01-18,2018-01-18 21:46:59,0,4
1,2018-03-10,2018-03-11 03:05:13,1,5
2,2018-02-17,2018-02-18 14:36:24,1,5
3,2017-04-21,2017-04-21 22:02:06,0,5
4,2018-03-01,2018-03-02 10:26:53,1,5


In [124]:
# Função pra agrupar em faixas de tempo de resposta
def faixa_resposta(dias):
    if dias == 0:
        return '1. No mesmo dia'
    elif dias <= 2:
        return '2. 1-2 dias'
    elif dias <= 7:
        return '3. 3-7 dias'
    else:
        return '4. Mais de 7 dias'

# Aplica a função na coluna de dias_resposta
df_reviews['faixa_resposta'] = df_reviews['dias_resposta'].apply(faixa_resposta)

# Calcula nota média e quantidade por faixa
nota_por_resposta = df_reviews.groupby('faixa_resposta').agg(
    nota_media=('review_score', 'mean'),
    quantidade=('review_score', 'count')
).reset_index()

# Calcula percentual
total = nota_por_resposta['quantidade'].sum()
nota_por_resposta['percentual'] = (nota_por_resposta['quantidade'] / total * 100).round(2)

# Arredonda a nota média
nota_por_resposta['nota_media'] = nota_por_resposta['nota_media'].round(2)

# Ordena pelas faixas
nota_por_resposta = nota_por_resposta.sort_values('faixa_resposta').reset_index(drop=True)

display(nota_por_resposta)

,faixa_resposta,nota_media,quantidade,percentual
0,1. No mesmo dia,3.89,24361,24.55
1,2. 1-2 dias,4.17,46893,47.26
2,3. 3-7 dias,4.13,23965,24.15
3,4. Mais de 7 dias,4.02,4005,4.04


In [125]:
# Gráfico de barras: nota média por faixa de tempo de resposta
fig_resposta = px.bar(
    nota_por_resposta,
    x='faixa_resposta',
    y='nota_media',
    title='Nota Média por Tempo de Resposta à Avaliação',
    labels={'faixa_resposta': 'Tempo de Resposta', 'nota_media': 'Nota Média (1 a 5)'},
    text='nota_media',
    template='plotly_white'
)

# Adiciona "⭐" depois do número em cima de cada barra
fig_resposta.update_traces(texttemplate='%{text}⭐', textposition='outside')

# Ajusta a escala do eixo Y pra evidenciar as diferenças
fig_resposta.update_yaxes(range=[3.5, 4.5])

fig_resposta.show()

**Tempo de resposta e satisfação:**

Quem responde a avaliação no mesmo dia da solicitação tende a dar nota mais baixa (3,89), enquanto quem responde em 1 a 2 dias avalia melhor (4,17). Isso sugere que clientes insatisfeitos têm maior motivação para responder imediatamente — quem teve experiência negativa quer reclamar logo. Já clientes satisfeitos respondem com mais calma, geralmente em 1-2 dias. A maioria (47%) está nessa faixa equilibrada.

###  Conclusão: Satisfação do Cliente

A análise de satisfação revelou três descobertas principais:

1. **Satisfação geral é boa, mas com cauda longa de insatisfação:** 77% dos clientes dão nota 4 ou 5, mas 11,5% dão nota mínima (1). Quem teve experiência muito ruim avalia com força, gerando um polo expressivo de insatisfação.

2. **Logística é o principal driver de satisfação:** a nota média cai de 4,03 (entregas no prazo) para 2,53 quando há qualquer atraso — uma queda de 37%. Em pedidos com mais de 10 dias de atraso, a nota despenca para 1,72. A boa notícia: 92% dos pedidos chegam antes do prazo prometido.

3. **Os meses de queda na satisfação coincidem com aumento de atrasos:** novembro/2017 (Black Friday) teve 12% de atrasos, fevereiro/2018 teve 14% e março/2018 atingiu 19% — os mesmos meses em que a nota média caiu. A operação não conseguiu absorver picos de volume sem comprometer a entrega.

Para sustentar o crescimento da Olist, **investir em capacidade logística para suportar picos de demanda** é o caminho mais direto para manter a satisfação alta.

##  Retenção e Recompra

Crescer em receita e ter alta satisfação é importante, mas há uma métrica ainda mais decisiva para o e-commerce: **o cliente volta a comprar?** Operações que dependem de aquisição constante de novos clientes são frágeis — toda parada de investimento em marketing derruba a receita. Esta análise investiga a base de clientes da Olist para entender o comportamento de recompra e o potencial de lifetime value.

### O dataset distingue cliente de pedido

Antes de calcular qualquer coisa, é importante entender uma sutileza importante da tabela `df_customers`:

- **`customer_id`** → identificador único **por pedido** (cada compra gera um novo)
- **`customer_unique_id`** → identificador único **da pessoa** (mesmo em todas as compras dela)

In [126]:
# Conta pedidos vs clientes únicos
total_pedidos = df_customers['customer_id'].nunique()
total_clientes_unicos = df_customers['customer_unique_id'].nunique()

print(f'Total de pedidos:                      {total_pedidos:,}')
print(f'Total de clientes únicos (pessoas):    {total_clientes_unicos:,}')
print(f'Pedidos a mais que clientes:           {total_pedidos - total_clientes_unicos:,}')
print()
print(f'Taxa de recompra (em pedidos):         {((total_pedidos - total_clientes_unicos) / total_pedidos * 100):.2f}%')

Total de pedidos:                      99,441
Total de clientes únicos (pessoas):    96,096
Pedidos a mais que clientes:           3,345

Taxa de recompra (em pedidos):         3.36%


In [127]:
# Conta quantos pedidos cada cliente único fez
pedidos_por_cliente = df_customers.groupby('customer_unique_id')['customer_id'].nunique().reset_index()
pedidos_por_cliente.columns = ['customer_unique_id', 'qtd_pedidos']

# Tabela completa de distribuição (documentação)
distribuicao_completa = pedidos_por_cliente['qtd_pedidos'].value_counts().reset_index()
distribuicao_completa.columns = ['num_compras', 'qtd_clientes']
distribuicao_completa = distribuicao_completa.sort_values('num_compras').reset_index(drop=True)

# Percentual
total = distribuicao_completa['qtd_clientes'].sum()
distribuicao_completa['percentual'] = (distribuicao_completa['qtd_clientes'] / total * 100).round(2)

# Percentual acumulado
distribuicao_completa['pct_acumulado'] = distribuicao_completa['percentual'].cumsum().round(2)

display(distribuicao_completa)

,num_compras,qtd_clientes,percentual,pct_acumulado
0,1,93099,96.88,96.88
1,2,2745,2.86,99.74
2,3,203,0.21,99.95
3,4,30,0.03,99.98
4,5,8,0.01,99.99
5,6,6,0.01,100.00
6,7,3,0.00,100.00
7,9,1,0.00,100.00
8,17,1,0.00,100.00


In [128]:
# Agrupa em categorias executivas (abrindo até 6, agrupando 7+)
def faixa_recompra(num):
    if num == 1:
        return '1 compra'
    elif num == 2:
        return '2 compras'
    elif num == 3:
        return '3 compras'
    elif num == 4:
        return '4 compras'
    elif num == 5:
        return '5 compras'
    elif num == 6:
        return '6 compras'
    else:
        return '7+ compras'

pedidos_por_cliente['faixa_recompra'] = pedidos_por_cliente['qtd_pedidos'].apply(faixa_recompra)

# Conta clientes por faixa
recompra_agrupada = pedidos_por_cliente['faixa_recompra'].value_counts().reset_index()
recompra_agrupada.columns = ['faixa', 'qtd_clientes']

# Calcula percentual
total = recompra_agrupada['qtd_clientes'].sum()
recompra_agrupada['percentual'] = (recompra_agrupada['qtd_clientes'] / total * 100).round(2)

# Ordena na ordem lógica
ordem = ['1 compra', '2 compras', '3 compras', '4 compras', '5 compras', '6 compras', '7+ compras']
recompra_agrupada['faixa'] = pd.Categorical(recompra_agrupada['faixa'], categories=ordem, ordered=True)
recompra_agrupada = recompra_agrupada.sort_values('faixa').reset_index(drop=True)

display(recompra_agrupada)

,faixa,qtd_clientes,percentual
0,1 compra,93099,96.88
1,2 compras,2745,2.86
2,3 compras,203,0.21
3,4 compras,30,0.03
4,5 compras,8,0.01
5,6 compras,6,0.01
6,7+ compras,5,0.01


In [129]:
import plotly.graph_objects as go

# Gráfico de barras: distribuição de recompra
fig_recompra = go.Figure()

fig_recompra.add_trace(
    go.Bar(
        x=recompra_agrupada['faixa'].astype(str),
        y=recompra_agrupada['qtd_clientes'],
        marker=dict(color='#000A53'),
        text=recompra_agrupada['percentual'],
        texttemplate='%{text}%',
        textposition='outside'
    )
)

fig_recompra.update_layout(
    title=dict(
        text='<b>Distribuição de Clientes por Número de Compras</b>',
        x=0.5,
        xanchor='center',
        font=dict(size=18)
    ),
    xaxis=dict(title='Número de Compras'),
    yaxis=dict(
        title=dict(text='Quantidade de Clientes', font=dict(color='#000A53')),
        tickfont=dict(color='#000A53'),
        tickformat=',.0f'
    ),
    template='plotly_white',
    showlegend=False,
    width=1200,
    height=600
)

fig_recompra.show()

**Retenção: o principal ponto de atenção da operação**

A análise da base de clientes revela o achado mais crítico de todo o relatório: a Olist é eficiente em atrair clientes, mas falha em retê-los. Dos quase 96 mil clientes únicos, a grande maioria realizou apenas uma compra. Somando quem comprou uma ou duas vezes, chega-se a praticamente toda a base.

A recompra é rara: clientes com três ou mais compras representam uma fração mínima do total. O cliente mais fiel da base realizou 17 compras — um caso isolado, provavelmente um pequeno revendedor que utiliza a plataforma de forma recorrente.

Esse padrão tem implicação direta para o negócio: todo o crescimento observado nas trilhas anteriores foi sustentado por aquisição de novos clientes, não por recompra da base existente. Em um cenário de aumento do custo de aquisição (CAC) ou desaceleração do investimento em marketing, a receita ficaria vulnerável.

Vale uma ressalva metodológica: por se tratar de um marketplace que opera através de outras plataformas (como Mercado Livre), parte da recompra pode não ser capturada — o mesmo cliente pode aparecer com identificadores diferentes. Ainda assim, o padrão geral de baixa retenção é consistente e merece atenção estratégica.

##  Trilha 5: Oportunidades e Recomendações

### Síntese das descobertas

A análise do dataset da Olist revelou uma operação em forte crescimento, com desafios pontuais que merecem atenção estratégica:

**Crescimento e Receita**
- Crescimento de 10x em pedidos e 9x em receita entre janeiro e novembro de 2017
- Estabilização em 2018 no patamar de 6.000 a 7.000 pedidos por mês
- Ticket médio mantido estável (R$ 124 a R$ 149), indicando que o crescimento veio do volume de clientes
- Forte concentração geográfica: o Sudeste responde por 65% da receita, enquanto Norte e Nordeste somam apenas 13,8% — apesar de apresentarem os maiores tickets médios (até R$ 182)

**Mix de Produtos**
- As 10 maiores categorias concentram 63% da receita, lideradas por Beleza & Saúde e Relógios & Presentes
- Categorias com lógicas distintas: algumas faturam por volume (ticket baixo), outras por valor agregado (ticket alto)

**Comportamento e Pagamentos**
- Cartão de crédito domina as transações (74%), seguido pelo boleto (19%)
- O brasileiro prefere parcelamentos curtos: 72% pagam à vista ou em até 4x
- Quanto maior o parcelamento, maior o ticket — clientes que parcelam em 10x gastam 4x mais que os que pagam à vista

**Satisfação e Retenção**
- 77% dos clientes dão notas 4 ou 5, mas 11,5% dão nota mínima
- 92% das entregas chegam antes do prazo; quando há atraso, a nota despenca de 4,03 para 2,53 (queda de 37%)
- Apenas 3% dos clientes voltam a comprar — a retenção é o principal ponto de atenção da operação

### Principais riscos identificados

A partir das análises, identificamos cinco riscos que merecem atenção estratégica da Olist:

**1. Baixa retenção de clientes**

Apenas 3% dos clientes realizam uma segunda compra. Todo o crescimento observado foi sustentado por aquisição de novos clientes, não por recompra da base existente. Esse é o risco mais relevante do negócio: em um cenário de aumento do custo de aquisição (CAC) ou desaceleração do investimento em marketing, a receita ficaria diretamente exposta.

**2. Vulnerabilidade logística em picos de demanda**

Os meses de maior volume (novembro/2017, fevereiro/2018 e março/2018) foram também os de maior taxa de atraso, chegando a 18,6% em março/2018. Como a satisfação é diretamente impactada por atrasos (queda de 37% na nota média), o crescimento futuro pode ser comprometido se a capacidade logística não escalar junto com o volume.

**3. Concentração geográfica**

O Sudeste concentra 65% da receita, com São Paulo respondendo sozinho por 38%. Essa dependência de uma única região representa risco em caso de saturação do mercado ou eventos regionais adversos. Ao mesmo tempo, é uma oportunidade: Norte e Nordeste somam apenas 13,8% da receita, mas têm os maiores tickets médios do país.

**4. Polarização das avaliações**

Apesar de 77% dos clientes avaliarem positivamente, 11,5% dão nota mínima — mais de 11 mil clientes muito insatisfeitos em uma base de quase 100 mil pedidos. Em mercados onde reputação online influencia decisões de compra, essa cauda longa de insatisfação pode prejudicar a aquisição de novos clientes.

**5. Dependência de um único canal de pagamento**

74% das transações dependem do cartão de crédito. Mudanças regulatórias, taxas de intercâmbio ou problemas com bandeiras impactariam três quartos da operação. A diversificação de meios de pagamento (como o Pix, indisponível no período do dataset) reduz esse risco.

### Recomendações estratégicas

A partir dos riscos identificados, propomos cinco recomendações prioritárias:

**1. Construir uma estratégia de retenção e recompra**

Com apenas 3% de recompra, há um espaço enorme a ser explorado. Ações possíveis: programa de fidelidade com benefícios crescentes, e-mail marketing pós-compra com recomendações personalizadas, cupons para a segunda compra e réguas de relacionamento segmentadas. Aumentar a recompra é a alavanca de maior impacto sobre a receita, já que reter custa menos que adquirir.

**2. Investir em capacidade logística escalável**

Como atrasos são o principal driver de insatisfação, a operação logística precisa escalar junto com o crescimento. Isso inclui ampliar parcerias com transportadoras, estabelecer hubs regionais e criar planos de contingência para datas críticas como a Black Friday. Manter a taxa de atraso abaixo de 5% deve ser uma meta operacional.

**3. Explorar a expansão para Norte e Nordeste**

Essas regiões representam apenas 13,8% da receita, mas têm os maiores tickets médios do país (até R$ 182). Investir em logística regional, parcerias locais e capilaridade pode destravar crescimento com receita unitária superior, ao mesmo tempo em que reduz a dependência do Sudeste.

**4. Aumentar o ticket médio com estratégias por categoria**

Como o crescimento veio de volume e não de ticket, vale diversificar a receita por cliente explorando o perfil de cada categoria. Categorias de ticket alto (Relógios & Presentes, Cool Stuff) são candidatas a ações de destaque e cross-sell; categorias de alto volume e ticket baixo (Cama, Mesa & Banho) pedem kits e combos para elevar o valor por pedido. Frete grátis acima de um valor mínimo também estimula carrinhos maiores.

**5. Diversificar meios de pagamento**

Reduzir a dependência do cartão de crédito (74%) ampliando alternativas como Pix, carteiras digitais e parcelamento via boleto. Além de reduzir riscos operacionais, isso diminui custos com taxas de intercâmbio.

### Conclusão final

A Olist apresentou um período de crescimento expressivo entre 2017 e 2018, com indicadores comerciais saudáveis e boa satisfação geral. A operação demonstrou capacidade de execução em escala, mas a análise revelou que esse crescimento foi sustentado quase inteiramente por aquisição de novos clientes — a recompra, de apenas 3%, é o principal ponto de atenção.

Os riscos identificados se concentram em cinco frentes endereçáveis: baixa retenção, capacidade logística sob pressão, concentração geográfica, polarização das avaliações e dependência do cartão de crédito. Nenhum deles é estrutural.

A boa notícia para investidores é que cada risco tem como contrapartida uma oportunidade clara: a baixa retenção é espaço para fidelização; a concentração no Sudeste convive com regiões de alto ticket ainda inexploradas; e o mix de categorias abre caminho para estratégias de cross-sell. Trata-se de uma operação fundamentalmente bem estruturada, com oportunidade de crescimento identificáveis e acionáveis.